### DISCUSSION TITLE
What Linkerd features will be broken if prometheus is down?

### DISCUSSION THREAD

**Discussion by hochuenw-dd** [linkerd/linkerd2#5710] [Q&A]
If users just don't install prometheus or prometheus is down, what features would they lose and still keep?

I suppose the mesh would still work, and based on a quick search in the source codes we would lose lots of observability features:
- linkerd dashboard
- linkerd edges 
- linkerd routes
- linkerd stat

The features that seems doesn't have hard dependency on prometheus are:
- linkerd diagnostics
- linkerd metrics
- linkerd profile 
- linkerd tap
- linkerd top

Is this true?

**Comment by kleimkuhler** [comment]
@hochuenw-dd Yes this is correct. I'm not sure if you've looked at the recent edge releases, but this separation of commands has actually been introduced by the Viz extension.

Right now the stable releases—most recent being `stable-2.9.3`—install all Linkerd components. If a user is not interested in commands such as `dashboard`, `edges`, `routes`, or `stat`, there is not a simple way to exclude the components those commands depend on from the install.

Now in the recent edges, `linkerd install` installs components related only to mTLS, load balancing, and other "core" functionalities.

`linkerd viz install` installs into a separate namespace and includes the Prometheus component, as well as tap and Grafana. Commands such as `edges`, `routes`, and `stat` how now moved under the `viz` subcommand and will not work until the Viz extension is installed.
